projects/components/src/download-json/download-json.component.ts

```diff
@@ -0,0 +1,59 @@
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+import { IconType } from '@hypertrace/assets-library';
+import { IconSize } from '@hypertrace/components';
+import { Observable } from 'rxjs';
+import { ButtonSize, ButtonStyle } from '../button/button';
+
+@Component({
+  selector: 'ht-download-json',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  styleUrls: ['./download-json.component.scss'],
+  template: `
+    <div class="download-json">
+      <ht-button
+        *ngIf="!this.dataLoading"
+        class="download-json"
+        icon="${IconType.Download}"
+        display="${ButtonStyle.Text}"
+        size="${ButtonSize.Large}"
+        [htTooltip]="this.tooltip"
+        (click)="this.triggerDownload()"
+      ></ht-button>
+      <ht-icon *ngIf="this.dataLoading" icon="${IconType.Loading}" size="${IconSize.Large}"></ht-icon>
+    </div>
+  `
+})
+export class DownloadJsonComponent {
```

Comment:
This seems like it could be easily renamed to be a generic download component

Reply:
As of now, this component does not support downloading other 'resources' like files etc. It only supports downloading a JavaScript data object as a JSON. Either we take care of all cases or rename this component to be more specific to its use case.

Reply:
It currently takes any string or object and supports downloading it as a file, which is the only download we'll likely have since we're using GQL. If we get new use cases in the future, can always expand supported inputs.

Reply:
In that case, i was thinking we can build a wrapper around this, which we can call 'ht-download'

Reply:
unnecessary, IMO. This is the only download mechanism we'll need, the only issue is that it won't always be json and that's already supported in the implementation as is.